include/powsybl/iidm/Connectable.hpp

```diff
@@ -54,6 +67,18 @@ class Connectable : public Identifiable {
 
     Terminal& getTerminal(unsigned long index);
 
+    void move(Terminal& oldTerminal, const std::string& oldConnectionInfo, const std::string& busId, bool connected);
+
+    void move(Terminal& oldTerminal, const std::string& oldConnectionInfo, unsigned long node, const std::string& voltageLevelId);
+
+private:
+    void attachTerminal(Terminal& oldTerminal, const std::string& oldConnectionInfo, VoltageLevel& voltageLevel, std::unique_ptr<Terminal>&& terminal);
+
+    friend class bus_terminal::BusBreakerViewImpl;
+
+    friend class node_terminal::BusBreakerViewImpl;
+    friend class node_terminal::NodeBreakerViewImpl;
```

Comment:
Not very proud of that... But this is necessary because `move` methods are not supposed to be in the public API...

Reply:
I totally agree: I wonder if the methods are well placed. Looking to the C++ implementation, you can access to the attach/detach methods of the voltage level from the different views. It's probably better to move the code there to avoid these friend declarations.

Could you please ensure it is feasible or not?

Reply:
To be discussed: I'm not sure to understand and/or I think we have an issue in C++ implementation: `attach` & `detach` methods are in the public API of the VoltageLevel in C++, and not in Java.